src/API/Orders.php

```diff
@@ -80,14 +80,77 @@ protected function prepare_objects_query( $request ) {
 		return $args;
 	}
 
+	/**
+	 * Get product IDs, names, and quantity from order ID.
+	 *
+	 * @param array $order_id ID of order.
+	 * @return array
+	 */
+	protected function get_products_by_order_id( $order_id ) {
+		global $wpdb;
+		$order_items_table    = $wpdb->prefix . 'woocommerce_order_items';
+		$order_itemmeta_table = $wpdb->prefix . 'woocommerce_order_itemmeta';
+		$products             = $wpdb->get_results(
+			$wpdb->prepare(
+				// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+				"SELECT
+				order_id,
+				order_itemmeta.meta_value as product_id,
+				order_itemmeta_2.meta_value as product_quantity,
+				order_itemmeta_3.meta_value as variation_id,
+				{$wpdb->posts}.post_title as product_name
+			FROM {$order_items_table} order_items
+			    LEFT JOIN {$order_itemmeta_table} order_itemmeta on order_items.order_item_id = order_itemmeta.order_item_id
+			    LEFT JOIN {$order_itemmeta_table} order_itemmeta_2 on order_items.order_item_id = order_itemmeta_2.order_item_id
+			    LEFT JOIN {$order_itemmeta_table} order_itemmeta_3 on order_items.order_item_id = order_itemmeta_3.order_item_id
+			    LEFT JOIN {$wpdb->posts} on {$wpdb->posts}.ID = order_itemmeta.meta_value
+			WHERE
+				order_id = ( %d )
+			    AND order_itemmeta.meta_key = '_product_id'
+				AND order_itemmeta_2.meta_key = '_qty'
+			  	AND order_itemmeta_3.meta_key = '_variation_id'
+			GROUP BY product_id
+			", // phpcs:enable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+				$order_id
+			),
+			ARRAY_A
+		);
```

Comment:
Get the products information by using the `order_itemmeta` table instead of the `wp_wc_order_product_lookup`, which doesn't rely on the scheduled actions.